react/week5/github-search-router/github-search-router/src/context/Context.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react';
+
+export const UserContext = React.createContext();
+
+export class Provider extends Component {
+   state = {
+      usersArray: [],
+      userName: '',
+   };
+
+   componentDidMount() {
+      this.findUser = (e) => {
+         e.preventDefault();
+         fetch(`https://api.github.com/search/users?q=${this.state.userName}`)
+            .then((res) => res.json())
+            .then((data) => {
+               console.log(data.items);
+               this.setState({
+                  usersArray: data.items,
+                  userName: '',
+               });
+            })
+            .catch((error) => console.log(error));
+      };
+   }
```

Comment:
I would avoid defining a method like this in `componentDidMount`. Is there a reason you didn't define this in the class directly?